helios-services/src/main/java/com/spotify/helios/agent/TaskRunner.java

```diff
@@ -151,20 +149,11 @@ private int run0() throws InterruptedException, DockerException {
           .setMaxIntervalMillis(SECONDS.toMillis(30))
           .build().newScheduler();
 
-      long totalMillis = 0;
-
       while (!healthChecker.get().check(containerId)) {
         final long retryMillis = retryScheduler.nextMillis();
         log.warn("container failed healthcheck, will retry in {}ms: {}: {}",
                  retryMillis, config, containerId);
-
-        totalMillis += retryMillis;
-        if (totalMillis >= HEALTHCHECK_MAX_RETRY_MILLIS) {
-          docker.killContainer(containerId);
-          throw new RuntimeException("container failed repeated health checks");
-        } else {
-          Thread.sleep(retryMillis);
-        }
+        Thread.sleep(retryMillis);
```

Comment:
Yea we can leave it for now and let helios-helper handle poll for RUNNING. If that doesn't happen within X time, we undeploy it and fail the build. We can add that logic here https://ghe.spotify.net/helios/helios-helper/blob/master/helios_helper/helios_deploy_job.py#L155